In [1]:
import pandas as pd
pd.options.mode.chained_assignment = None  # default='warn'
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from scipy.stats import pearsonr
from data_analysis import get_hyp1_data, plot_data, calc_corr

## Hypothesis 1: The hike in the price of food for developing countries is low as compared to underdeveloped countries.

### Read the food prices dataset

In [2]:
combined_data = get_hyp1_data()
combined_data

country_name          commodity_name  month  year  price  \
0       Afghanistan                   Bread      1  2014   50.0   
1       Afghanistan                   Bread      2  2014   50.0   
2       Afghanistan                   Bread      3  2014   50.0   
3       Afghanistan                   Bread      4  2014   50.0   
4       Afghanistan                   Bread      5  2014   50.0   
...             ...                     ...    ...   ...    ...   
743909  South Sudan                  Sesame      5  2017  400.0   
743910  South Sudan           Fuel (diesel)      2  2017   12.5   
743911  South Sudan           Fuel (diesel)      3  2017   95.0   
743912  South Sudan  Fuel (petrol-gasoline)      2  2017   95.0   
743913  South Sudan  Fuel (petrol-gasoline)      3  2017   93.7   

        country_status  price_factor  usd_price  
0           Developing        0.0100    0.50000  
1           Developing        0.0100    0.50000  
2           Developing        0.0100    0.50000  
3           Developing        0.0100    0.50000  
4           Developing        0.0100    0.50000  
...                ...           ...        ...  
743909  Underdeveloped        0.0023    0.92000  
743910  Underdeveloped        0.0023    0.02875  
743911  Underdeveloped        0.0023    0.21850  
743912  Underdeveloped        0.0023    0.21850  
743913  Underdeveloped        0.0023    0.21551  

[743914 rows x 8 columns]

### Pearson correlation coefficient and p-value for testing non-correlation

<font size = '3'>The Pearson correlation coefficient measures the linear relationship between two datasets. The calculation of the p-value relies on the assumption that each dataset is normally distributed. Like other correlation coefficients, this one varies between -1 and +1 with 0 implying no correlation. Correlations of -1 or +1 imply an exact linear relationship. Positive correlations imply that as x increases, so does y. Negative correlations imply that as x increases, y decreases.

The p-value roughly indicates the probability of an uncorrelated system producing datasets that have a Pearson correlation at least as extreme as the one computed from these datasets. (https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.pearsonr.html)</font>

## Now, lets compare the price of Bread for Kenya (Developing) and Gambia (Underdeveloped)

In [3]:
plot_data(combined_data, "Kenya", "Gambia", "Bread")
calc_corr(combined_data, "Kenya", "Gambia", "Bread")

NameError: name 'selective_data' is not defined

<font size = '3'>Here, Although the graphs have a positive slope from the years 2013 to 2015, we can observe a negative slope in the price of Bread in Kenya for the year 2015-2016. Furthermore, the correlation coefficient is 0.08. As the correlation coefficient is close to zero, we can conclude that there is no correlation between the price of bread between the two countries.</font>

# To conduct further analysis, let us test four more countries for the same commodity i.e Bread.
### First we consider, Tajikistan (Developing) and Guinea (Underdeveloped)

In [ ]:
plot_data(combined_data, "Tajikistan", "Guinea", "Bread")
calc_corr(combined_data, "Tajikistan", "Guinea", "Bread")

### Tajikistan (Developing) and Kyrgyzstan (Underdeveloped)

In [ ]:
plot_data(combined_data, "Tajikistan", "Kyrgyzstan", "Bread")
calc_corr(combined_data, "Tajikistan", "Kyrgyzstan", "Bread")

<font size = '3'>Lets consider the first graph i.e Tajikistan (Developing) and Guinea (Underdeveloped):
    
The price of Bread in Tajikistan has observed slight up and down variation over the years. On the other hand, the price of Bread in Guinea reached a peak of ~0.32 and a bottom of ~0.26 over the same period.

Moving on to the next graph i.e Tajikistan (Developing) and Kyrgyzstan (Underdeveloped):

Kyrgyzstan has a steep positive slope from the years 2013 to 2015. For the same period, Tajikistan also has a positive slope, but the value of slope is comparatively less.

Lastly, the correlation coefficient for the two graphs is -0.16 and 0.75 respectively. Although there is a possibility of correlation between the price of Bread for Tajikistan (Developing) and Kyrgyzstan (Underdeveloped) in the future, the observed variation and uncertainty visible in the three graphs clearly indicate that the increase/decrease of bread prices is unrelated.</font>


# Now, to further examine the trend, lets test our analysis for a different commodity, i.e Wheat

### Comparing price of Wheat for Afghanistan (Developing) and Pakistan (Underdeveloped)

In [ ]:
plot_data(combined_data, "Afghanistan", "Pakistan", "Wheat")
calc_corr(combined_data, "Afghanistan", "Pakistan", "Wheat")

<font size = '3'>For the initial years of the above graph, the lines follow a same trend, which aligns to our said hypothesis. But if we observe the year 2015-2016, price of Wheat increases in Afghanistan whereas it stays the same in Pakistan. Lastly, for the year 2016-2017, the price of wheat follows an opposite trend to what was seen earlier. The commodity price decreases in Afghanistan whereas it increases in Pakistan. Lastly, the correlation coefficient is 0.3, therefore the price of wheat for the two countries are not related.</font>

### Comparing price of Wheat for Nepal (Developing) and Ethiopia (Underdeveloped)

In [ ]:
plot_data(combined_data, "Nepal", "Ethiopia", "Wheat")
calc_corr(combined_data, "Nepal", "Ethiopia", "Wheat")

### Comparing price of Wheat for India (Developing) and Pakistan (Underdeveloped)

In [ ]:
plot_data(combined_data, "India", "Pakistan", "Wheat")
calc_corr(combined_data, "India", "Pakistan", "Wheat")

<font size = '3'>Let us consider the first graph i.e Nepal (Developing) and Ethiopia (Underdeveloped):
    
On one hand the price of Wheat was highly stable in Nepal while on the other hand the price of wheat in Ethiopia observed several ups and downs.

Similarly, in the above graph i.e India (Developing) and Pakistan (Underdeveloped):
    
The two countries follow a similar trend for the year 2013-2014 but for the year 2014-2015 they follow an opposite pattern. For the year 2014-2015, the price of wheat in India goes from ~0.27 USD to ~0.28 USD whereas in Pakistan the same goes from ~0.23 USD to ~0.19 USD. This clearly indicates that the food prices in the two countries are not related to each other.
Laslty, the correlation coefficient for the two graphs is -0.17 and 0.22 respectively, this confirms our observation.</font>

### From the above analysis, it is clearly evident that the food prices in Developing and Underdeveloped countries are not related.
# Therefore we reject our hypothesis